### PR TITLE
fix: use null type when read from unknown row

### LINKED ISSUE #12120: `from_dicts` assigns boolean dtype to undeclared columns
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
data = [{"a": 1, "b": 2}]
result = pl.from_dicts(data, schema=["x"])
print(result)
```

### Log output

```shell
shape: (1, 1)
┌──────┐
│ x    │
│ ---  │
│ bool │
╞══════╡
│ null │
└──────┘
```

### Issue description

The description of the `schema` parameter for `from_dicts` is extensive, but it's not exactly clear what to expect in this situation. In any case, a boolean dtype is unexpected here. I would expect a `Float32`, which is the default for undeclared columns without values.

### Expected behavior

Column of dtype `Float32`.

```python
import polars as pl
from polars.testing import assert_frame_equal

data = [{"a": 1, "b": 2}]
result = pl.from_dicts(data, schema=["x"])
expected = pl.DataFrame({"x": [None]})
assert_frame_equal(result, expected)
```

### Installed versions

main branch

### PR DESCRIPTION
**Pull request by reswqa**
This closes #12120.

### CODE REVIEW

File: py-polars/src/dataframe.rs [hunk 1]
```diff
@@ -56,10 +56,10 @@ impl PyDataFrame {
         let schema =
             rows_to_schema_supertypes(&rows, infer_schema_length.map(|n| std::cmp::max(1, n)))
                 .map_err(PyPolarsErr::from)?;
-        // Replace inferred nulls with boolean and erase scale from inferred decimals.
+        // Replace inferred nulls with f32 and erase scale from inferred decimals.
         let fields = schema.iter_fields().map(|mut fld| match fld.data_type() {
             DataType::Null => {
-                fld.coerce(DataType::Boolean);
```
[the remark this answers is not in the document]
Comment (stinodego) [reply py-polars/src/dataframe.rs:62]: That's what I thought as well. But if this is in the code explicitly, maybe it's there for a reason? 🤔 

Maybe @ritchie46 can chime in.

**Review by stinodego — APPROVED** [review]
Looks good to me, though maybe Ritchie can chime in whether this existing boolean behavior is there for a reason.

**Comment by alexander-beedie** [comment]
> Looks good to me, though maybe Ritchie can chime in whether this existing boolean behavior is there for a reason.

I'm curious why, if given a column of Nulls, we don't declare it as... a Null column? 🤔

**Comment by stinodego** [comment]
> But I think this is consistent with the following behavior

Indeed, though I think that behavior stems from before we had the `Null` type. Maybe it's better to just set this to `Null`?

**Comment by ritchie46** [comment]
@reswqa can you do a rebase?

**Review by ritchie46 — APPROVED** [review]
Thank you @reswqa. :+1:

**Comment by stinodego** [comment]
> @stinodego Do you think we should similarly change the behavior of `s = pl.Series([None])`, so that it also has `null` dtype instead of `f32`. If yes, I will take a look.

@reswqa Yes, that would be great!

**Pull request merged**